src/components/progress-circle/progress-circle.ts

```diff
@@ -41,48 +84,100 @@ export class MdProgressCircle {
    * mode is bound to the host as the attribute host.
    */
   @HostBinding('attr.mode')
-  @Input() mode: 'determinate' | 'indeterminate' = 'determinate';
+  @Input()
+  get mode() {
+    return this._mode;
+  }
+  set mode(m: ProgressCircleMode) {
+    if (m == 'indeterminate') {
+      this._startIndeterminateAnimation();
+    } else {
+      this._cleanupIndeterminateAnimation();
+    }
+    this._mode = m;
+  }
+  private _mode: ProgressCircleMode = 'determinate';
+
+  constructor(changeDetectorRef: ChangeDetectorRef) {
+    this._changeDetectorRef = changeDetectorRef;
+  }
 
 
   /**
-   * Gets the current stroke dash offset to represent the progress circle.
-   *
-   * The stroke dash offset specifies the distance between dashes in the circle's stroke.
-   * Setting the offset to a percentage of the total circumference of the circle, fills this
-   * percentage of the overall circumference of the circle.
+   * Animates the rendered circle.
    */
-  strokeDashOffset() {
-    // To determine how far the offset should be, we multiple the current percentage by the
-    // total circumference.
+  private _renderCircle(animateFrom: number, animateTo: number,
+                        ease: (currentTime: number, startValue: number,
+                               changeInValue: number, duration: number) => number,
```

Comment:
You should be able to alias this type so that it's a little more readable here:
```ts
type EasingFn = (currentTime: number, startValue: number, changeInValue: number, duration: number) => number
```
(see the `Dir` directive for an example)